tests/test_jx/test_deep_ops.py

```diff
@@ -356,9 +356,9 @@ def test_deep_names_w_star(self):
             "expecting_list": {
                 "meta": {"format": "list"},
                 "data": [
-                    {"a": {"_t": {"b.s": 1, "h.s": "a-a"}}},
-                    {"a": {"_t": {"b.s": 2, "h.s": "a-b"}}},
-                    {"a": {"_t": {"b.s": 3, "h.s": "a-c"}}}
+                    {"a._t.b.s": 1, "a._t.h.s": "a-a"},
+                    {"a._t.b.s": 2, "a._t.h.s": "a-b"},
+                    {"a._t.b.s": 3, "a._t.h.s": "a-c"}
```

Comment:
I believe this is wrong, but maybe so is the original:
* `"select": ["a._t"]` (with array) would result in list like `[{"a": {"_t": blah}}, ...]` because the dots are not escaped.
* `"select": "a._t"` (without array) is a list of values, like `[blah, ...]` because values have no names
* `"select": "a._t.*" should look like `[{"b.s": 1, "h.s": "a-a"}, ...]` because it is not an array

Reply:
The table and cube headers may also be wrong; you revealed an inconsistency with the naming and the escaping of dots:

    "header": ["a._t.b.s", "a._t.h.s"],

should read 

    "header": ["a._t.b\.s", "a._t.h\.s"],

Reply:
Following the pattern of `*` from other tests in the select clause, I think `"select": "a._t.*"` should give us `{"a._t.b.s": 1, "a._t.h.s": "a-a"}` .

Reply:
test: `test_select_w_deep_star` shows the concern.

Reply:
I added a document to [explain the correct star formatting](https://github.com/mozilla/jx-sqlite/blob/dev/docs/jx_expressions_leaves.md).  Please look it over, and we can discuss how the tests, and this document disagree. Hopefully this (new) definition will simplify the code.

Reply:
made some changes because I do not like the conclusions

https://github.com/mozilla/jx-sqlite/commit/5dcd147453c5cc12e9702bdcba2515ff84e7e3cd